agent/src/beerocks/slave/tasks/task.h

```diff
@@ -48,8 +48,9 @@ class Task {
      * @brief Handle events, which are being send from containing thread.
      * 
      * @param event_enum_value Event enum value which shall be defined on the task.
+     * @param event_obj Pointer to some chunk of memory used to pass data to the event handler.
      */
-    virtual void handle_event(uint8_t event_enum_value){};
+    virtual void handle_event(uint8_t event_enum_value, void *event_obj){};
```

Comment:
lets say you have 2 ap_autoconfiguration tasks in the unified agent running in parallel , 
how do you make sure they don't write to the same "chunk of memory"
why not use AgentDB?

Reply:
1. There is no such thing task of the same type running in parallel on the Agent.
2. Even if it was allowed to run several instances of the same task, they would not run in parallel. They would run one after another.
3. The chunk of memory is a way to pass some type of object without explicitly declare it on the handle_event signature,
the only difference is that it passed as void pointer instead of explicit type.
Either way, the incoming chuck of memory is for reading, and not writing.

Reply:
> They would run one after another. 
that's exactly what i mean and now you are allowing them to use shared resource without protecting it.
>    Either way, the incoming chuck of memory is for reading, and not writing.
the user doesn't know its not for writing. you can't make everybody read instructions all the time.

If you must pass it as a const.

Reply:
This is not a shared resource, only one task will get it.

Reply:
Not true since you are adding it to the task class.

Reply:
We agreed that it is a valid change in a private conversation, therefore mark as resolve.